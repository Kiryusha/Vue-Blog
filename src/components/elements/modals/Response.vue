<template lang="pug">
  modal(
    name="response"
    width="100%"
    height="auto"
    @before-open="beforeOpen"
  )
    ModalWrapper
      .modal__content
        .modal__title {{title}}
        .modal__close
          Button(
            @click="$modal.hide('response')"
          ) Закрыть
</template>

<script>
import ModalWrapper from 'Components/elements/modals/Modal';
import Button from 'Components/elements/general/Button';

export default {
  components: {
    ModalWrapper,
    Button,
  },
  data() {
    return {
      title: '',
    };
  },
  methods: {
    beforeOpen(event) {
      this.title = event.params.message;
    },
  },
};
</script>
