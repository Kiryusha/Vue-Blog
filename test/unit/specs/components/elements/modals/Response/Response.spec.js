// Response.vue

import { shallowMount } from '@vue/test-utils';
import Response from 'Components/elements/modals/Response.vue';

describe('Response.vue', () => {
  const config = {
    stubs: ['modal'],
  };

  describe('component shallow rendering', () => {
    it('should match a snapshot', () => {
      const wrapper = shallowMount(Response, config);

      expect(wrapper.html()).toMatchSnapshot();
    });
  });

  describe('custom actions', () => {
    it('should set title data property on beforeOpen()', () => {
      const wrapper = shallowMount(Response, config);

      wrapper.vm.beforeOpen({
        params: {
          message: 'test',
        }
      })

      expect(wrapper.vm.title).toBe('test');
    });
  });
});
