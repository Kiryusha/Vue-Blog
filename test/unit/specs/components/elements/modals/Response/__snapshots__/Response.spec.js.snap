// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Response.vue component shallow rendering should match a snapshot 1`] = `<modal-stub name="response" width="100%" height="auto"></modal-stub>`;
